### PR TITLE
max_unpool1d #19547

### LINKED ISSUE #19547: max_unpool1d

### PR DESCRIPTION
**Pull request by erimdogan**
close #19547

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 1]
```diff
@@ -37,6 +37,37 @@ def test_max_pool1d(
...
+        fw=backend_fw,
+        on_device=on_device,
+        fn_name=fn_name,
+        rtol_=1e-2,
+        atol_=1e-2,
+        x=x[0],
+        kernel=kernel,
+        strides=stride,
+        padding=pad,
+        indices=indices
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:69]:
```
-        indices=indices
+        indices=indices,
```

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 2]
```diff
@@ -37,6 +37,37 @@ def test_max_pool1d(
...
+    )
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:71]: please add one extra empty line here

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 3]
```diff
@@ -37,6 +37,37 @@ def test_max_pool1d(
         padding=pad,
     )
 
+@handle_test(
+        fn_tree="functional.ivy.experimental.layers.max_unpool1d",
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:41]:
```
-        fn_tree="functional.ivy.experimental.layers.max_unpool1d",
-        x_k_s_p=helpers.arrays_for_pooling(min_dims=3, max_dims=3, min_side=1, max_side=4),
-        indices = st.lists(st.integers(0,1), min_size=1, max_size=4), 
-        ground_truth_backend="jax",
-        test_gradients=st.just(False),  
+    fn_tree="functional.ivy.experimental.layers.max_unpool1d",
+    x_k_s_p=helpers.arrays_for_pooling(min_dims=3, max_dims=3, min_side=1, max_side=4),
+    indices=st.lists(st.integers(0, 1), min_size=1, max_size=4),
+    ground_truth_backend="jax",
+    test_gradients=st.just(False),
```

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 4]
```diff
@@ -37,6 +37,37 @@ def test_max_pool1d(
         padding=pad,
     )
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:39]: please add one extra empty line here

File: ivy/functional/ivy/experimental/layers.py [hunk 5]
```diff
@@ -88,6 +88,75 @@ def max_pool1d(
...
+    >>> print(unpool_result)
+    ivy.array([[[ 0.,  4.,  0.,  5.,  0.,  6.,  0.,  7.,  0.,  0.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  8.,  0.,  9.,  0., 10.,  0., 11.,  0.]],
+
+           [[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0., 16.,  0., 17.,  0.],
+            [ 0., 18.,  0., 19.,  0.,  0.,  0.,  0., 20.,  0., 21.,  0.]]])
+    """
+    return ivy.current_backend(x).max_unpool1d(
+        x, indices, kernel, strides, padding, data_format=data_format, out=out
+)
```
Comment (Daniel4078) [inline ivy/functional/ivy/experimental/layers.py:158]:
```
-)
+    )
```

File: ivy/functional/ivy/experimental/layers.py [hunk 6]
```diff
@@ -88,6 +88,75 @@ def max_pool1d(
         x, kernel, strides, padding, data_format=data_format, out=out
     )
 
+@handle_nestable
+@handle_out_argument
+@to_native_arrays_and_back
+@handle_device_shifting
+def max_unpool1d(
+    x:  ivy.Union[ivy.Array, ivy.NativeArray],
```
Comment (Daniel4078) [inline ivy/functional/ivy/experimental/layers.py:96]:
```
-    x:  ivy.Union[ivy.Array, ivy.NativeArray],
-    indices : Union[ivy.Array, ivy.NativeArray],
-    kernel : Union[int, Tuple[int]],
-    strides : Union[int, Tuple[int]],
-    padding : str,
+    x: ivy.Union[ivy.Array, ivy.NativeArray],
+    indices: Union[ivy.Array, ivy.NativeArray],
+    kernel: Union[int, Tuple[int]],
+    strides: Union[int, Tuple[int]],
+    padding: str,
     /,
     *,
-    data_format : str = "NWC",
-    out : Optional[ivy.Array] = None
+    data_format: str = "NWC",
+    out: Optional[ivy.Array] = None,
```

File: ivy/functional/ivy/experimental/layers.py [hunk 7]
```diff
@@ -88,6 +88,75 @@ def max_pool1d(
         x, kernel, strides, padding, data_format=data_format, out=out
     )
```
Comment (Daniel4078) [inline ivy/functional/ivy/experimental/layers.py:90]: please add one extra empty line here

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello, I found some formatting issues that you need to fix. Please check out the comments below, thank you!

**Comment by erimdogan** [comment]
I made the changes that you reccommend. Thanks for advices.

**Review by Daniel4078 — APPROVED** [review]
Looks good to me!

**Review by Daniel4078 — COMMENTED** [review]
Well, it seems like there is no backend implemenations written for max_unpool1d so the tests are not passing

**Pull request merged**